crates/nu-protocol/src/value/test_derive.rs

```diff
@@ -494,3 +494,24 @@ fn bytes_roundtrip() {
         .into_test_value();
     assert_eq!(expected, actual);
 }
+
+#[test]
+fn struct_type_name_attr() {
+    #[derive(FromValue, Debug)]
+    #[nu_value(type_name = "struct")]
+    struct TypeNameStruct;
+
+    assert_eq!(
+        TypeNameStruct::expected_type().to_string().as_str(),
+        "struct"
+    );
+}
+
+#[test]
+fn enum_type_name_attr() {
+    #[derive(FromValue, Debug)]
+    #[nu_value(type_name = "enum")]
+    struct TypeNameEnum;
```

Comment:
That may be a slightly confusing example implying that it would generate an enum.